### DISCUSSION TITLE
Issues building native image with dockerBuildNative or nativeCompile in serverless platform

### DISCUSSION THREAD

**Discussion by dliang5** [micronaut-projects/micronaut-core#11092] [Q&A]
Hey all! 

I'm working on migrating a pretty small project from Micronaut 2 to 4.5.1. It honestly doesn't contain much just your standard CRUD application with 1 controller with 4 endpoints and some files setup to connect to GCP Cloud SQL. 

Once migrated to 4.5.1, I'm able to run all the commands to run this application locally using the gradle plugin like `dockerBuildNative`, `dockerBuild`, `nativeCompile`, `nativeRun`, `run`, etc. Compilation times for native-image takes like 5-8 mins with Apple silicon. My issue is when I build this using Bamboo, Bitbucket Pipelines, or GCP Build it kind of hangs and I would cancel my run after 60 mins or an OOM error message shows up. The setup for these serverless platforms are 8 GB and 2 CPU threads if that matters

Is executing the `native-image` command ideal in a serverless CI setting? Or is there another set of best practices for this? Unfortunately, I don't have another person at work to bounce ideas with Micronaut as all the original devs have left over the years haha. I can also show my build.gradle file but it's quite literally a basic CRUD application that we only run in Cloud Run.

**Comment by sdelamo** [comment]
yes, building a native image in CI is common use case .

**Reply by dliang5** [reply]
Hey @sdelamo , thanks for answering. Do you happen to have any clues on my other issue where native image build seemingly hangs when building and goes beyond 60 mins using `dockerBuildNative` command in CI? Prior to upgrading to micronaut 4 version, we had a simple Dockerfile that just copied the libs, resources, and application.jar without excluding any configs that worked fine (took 12 mins on avg to build).

**Reply by sdelamo** [reply]
No idea, why your build is hanging.
